pygen/pygen_src/isa/riscv_b_instr.py

```diff
@@ -0,0 +1,178 @@
+"""
+Copyright 2020 Google LLC
+Copyright 2020 PerfectVIPs Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+"""
+
+import math
+import logging
+import vsc
+from importlib import import_module
+from pygen_src.isa.riscv_instr import riscv_instr
+from pygen_src.riscv_instr_pkg import (pkg_ins, riscv_instr_category_t, riscv_reg_t,
+                                       riscv_instr_name_t, riscv_instr_group_t,
+                                       riscv_instr_format_t)
+from pygen_src.riscv_instr_gen_config import cfg
+rcs = import_module("pygen_src.target." + cfg.argv.target + ".riscv_core_setting")
+
+
+@vsc.randobj
+class riscv_b_instr(riscv_instr):
+    def __init__(self):
+        super().__init__()
+        self.rs3 = vsc.rand_enum_t(riscv_reg_t)
+        self.has_rs3 = vsc.bit_t(1)
+
+    def set_rand_mode(self):
+        super().set_rand_mode()
+        self.has_rs3 = 0
+        if self.format == riscv_instr_format_t.R_FORMAT:
+            if self.instr_name in [riscv_instr_name_t.CLZW,
+                                   riscv_instr_name_t.CTZW, riscv_instr_name_t.PCNTW,
+                                   riscv_instr_name_t.SEXT_B, riscv_instr_name_t.SEXT_H,
+                                   riscv_instr_name_t.CLZ, riscv_instr_name_t.CTZ,
+                                   riscv_instr_name_t.PCNT, riscv_instr_name_t.BMATFLIP,
+                                   riscv_instr_name_t.CRC32_B, riscv_instr_name_t.CRC32_H,
+                                   riscv_instr_name_t.CRC32_W, riscv_instr_name_t.CRC32C_B,
+                                   riscv_instr_name_t.CRC32C_H, riscv_instr_name_t.CRC32C_W,
+                                   riscv_instr_name_t.CRC32_D, riscv_instr_name_t.CRC32C_D]:
+                self.has_rs2 = 0
+        elif self.format == riscv_instr_format_t.R4_FORMAT:
+            self.has_imm = 0
+            self.has_rs3 = 1
+        elif self.format == riscv_instr_format_t.I_FORMAT:
+            self.has_rs2 = 0
+            if self.instr_name in [riscv_instr_name_t.FSRI, riscv_instr_name_t.FSRIW]:
+                self.has_rs3 = 1
+
+    def pre_randomize(self):
+        super().pre_randomize()
+        with vsc.raw_mode():
+            self.rs3.rand_mode = bool(self.has_rs3)
+
+    def set_imm_len(self):
+        if self.format == riscv_instr_format_t.I_FORMAT:
+            if self.category in [riscv_instr_category_t.SHIFT, riscv_instr_category_t.LOGICAL]:
+                if (self.group.name == riscv_instr_group_t.RV64B and
+                        not(self.instr_name in [riscv_instr_name_t.SLLIU_W])):
```

Comment:
since the list has only one member, I think it's better for this line to be replaced with:
`self.instr_name != riscv_instr_name_t.SLLIU_W`